app/kuma-dp/pkg/dataplane/command/base_command.go

```diff
@@ -0,0 +1,27 @@
+package command
+
+import (
+	"context"
+	"io"
+	"os/exec"
+	"syscall"
+	"time"
+)
+
+func baseBuildCommand(
+	ctx context.Context,
+	stdout io.Writer,
+	stderr io.Writer,
+	name string,
+	args ...string,
+) *exec.Cmd {
+	command := exec.CommandContext(ctx, name, args...)
+	command.Stdout = stdout
+	command.Stderr = stderr
+	command.Cancel = func() error {
+		return command.Process.Signal(syscall.SIGTERM)
+	}
+	command.WaitDelay = time.Second * 5
```

Comment:
why do we need this? is this 5 second associated with any other value that we wait on?

Reply:
So this is the time between a context cancelled and clean shutdown to be issued and a Kill if the process is still around.
It's a fail safe to make sure processes don't leak. I don't think it's worth making it configurable yet as it's very much just a fail safe.